gems/aws-sigv4/sig/signer.rbs

```diff
@@ -0,0 +1,70 @@
+module Aws
+  module Sigv4
+    class Signer
+      def initialize: (
+        service: String,
+        region: String,
+        access_key_id: String,
+        secret_access_key: String,
+        session_token?: String?,
+        unsigned_headers?: Array[String]?,
+        uri_escape_path?: bool,
+        apply_checksum_header?: bool,
+        signing_algorithm?: Symbol,
+        omit_session_token?: bool,
+        normalize_path?: bool,
+      ) -> void
+      | (
+        service: String,
+        region: String,
+        credentials: Credentials,
+        unsigned_headers?: Array[String]?,
+        uri_escape_path?: bool,
+        apply_checksum_header?: bool,
+        signing_algorithm?: Symbol,
+        omit_session_token?: bool,
+        normalize_path?: bool,
+      ) -> void
+      | (
+        service: String,
+        region: String,
+        credentials_provider: Object,
+        unsigned_headers?: Array[String]?,
+        uri_escape_path?: bool,
+        apply_checksum_header?: bool,
+        signing_algorithm?: Symbol,
+        omit_session_token?: bool,
+        normalize_path?: bool,
+      ) -> void
```

Comment:
I think you are writing the overload to match the YARD notation, but there is a tradeoff with maintenance costs.
Either way, we can't make a strict type determination when dealing with hash arguments.

Reply:
I've tried combining all the overloads of the initialize method into a single definition. Would you mind sharing your thoughts on this approach?
2a780e6cc1c843a179acdebeb5382297ca390aba

Reply:
Hash arguments should be supported.

Reply:
Let me confirm: Do you mean I should revise it as follows?

```ruby
def initialize: (
  options: {
    service: String,
    region: String,
    access_key_id?: String,
    secret_access_key?: String,
    session_token?: String,
    credentials?: untyped,
    credentials_provider?: untyped,
    unsigned_headers?: Array[String],
    uri_escape_path?: bool,
    apply_checksum_header?: bool,
    signing_algorithm?: :sigv4 | :sigv4a | :'sigv4-s3express',
    omit_session_token?: bool,
    normalize_path?: bool,
  }
) -> void
```

Reply:
No. "Hash arguments" is same as https://github.com/aws/aws-sdk-ruby/pull/3152#discussion_r1870488007

How are you testing the behavior?
I don't think it will pass in steep with the proposed signature.

Reply:
Sorry about that, I've updated the code.
3c34687562bbc42d3dbbf4372b96410fd5d32a83